yellowbrick/classifier.py

```diff
@@ -50,6 +51,149 @@ def __init__(self, model, ax=None, **kwargs):
+        #Initialize all the other attributes we'll use (for coder clarity)
+        self.confusion_matrix = None
+
+        self.cmap = color_sequence(kwargs.pop('cmap', 'YlOrRd'))
+
+        #If possible, assign the classes. If this can't happen now, it will happen in the .fit() method
+        #TODO should consider implementing this approach in ClassificationReport - allows the user to use
+            #the default classes_ list without calling fit (i.e. if the model has already been fit before
+            #initializing the visualizer)
+        if classes is None:
+            try:
+                self.classes_ = self.estimator.classes_
+            except AttributeError:
+                self.classes_ = None
+        else:
+            self.classes_ = classes
+
+    #TODO this is the same as ClassificationReport, should hoist it up a level
+    #TODO is there any way to get this documentation to extend the parent class documentation so we don't have to double(triple, quadruple...) maintain it?
+    def fit(self, X, y=None, **kwargs):
+        """
+        Parameters
+        ----------
+
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs: keyword arguments passed to Scikit-Learn API.
+        """
+        super(ConfusionMatrix, self).fit(X, y, **kwargs)
+        if self.classes_ is None:
+            self.classes_ = self.estimator.classes_
+        return self
+
+    def score(self, X, y, sample_weight=None):
+        """
+        Generates the Scikit-Learn confusion_matrix and applies this to the appropriate axis
+
+        Parameters
+        ----------
+
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        sample_weight: optional, passed to the confusion_matrix
+        """
+        y_pred = self.predict(X)
+        self.confusion_matrix = confusion_matrix(y_true = y, y_pred = y_pred, labels=self.classes_, sample_weight=sample_weight)
+        #Convert confusion matrix to percent of true
+        self.confusion_matrix = np.round(self.confusion_matrix / np.sum(self.confusion_matrix, axis=1) * 100, decimals=0)
+        self.confusion_matrix = self.confusion_matrix[::-1]
+
+        return self.draw()
+
+    def draw(self):
+        # Create the axis if it doesn't exist
+        if self.ax is None:
+            self.ax = plt.gca()
+
+
+
+        #fig = plt.imshow(self.confusion_matrix, interpolation='nearest', cmap=self.cmap, vmin=0, vmax=1)
+        X = np.linspace(start=0, stop=len(self.classes_), num=len(self.classes_))
+        Y = np.linspace(start=0, stop=len(self.classes_), num=len(self.classes_))
+
+        # Draw the heatmap
+        mesh = self.ax.pcolormesh(X,Y,self.confusion_matrix, vmin=0, vmax=100,
```

Comment:
Does `pcolormesh` handle light text on dark background and dark text on light background? If so, we should definitely move to it. But also in my reading, I'm seeing that it can handle larger matrices; so I think it's a good idea to use it over `imshow`.